### PR TITLE
fix: revert default value for trie cache capacity

### PR DESCRIPTION
**Pull request by Longarithm**
Revert default value (to 50K) because after https://github.com/near/nearcore/pull/7022 we got more evidence that it doesn't help to speed up storage ops.

## Testing

Existing tests.

### CODE REVIEW

File: core/store/src/config.rs [hunk 1]
```diff
@@ -97,7 +97,7 @@ impl Default for StoreConfig {
             // we use it since then.
             block_size: bytesize::ByteSize::kib(16),
 
-            trie_cache_capacities: vec![(ShardUId { version: 1, shard_id: 3 }, 2_000_000)],
```
Comment (matklad) [inline core/store/src/config.rs:100]: Do we want to keep the functionality to override this per shard?

Comment (matklad) [reply core/store/src/config.rs:100]: I am worried about "per-shard" bit. From user's peerspective, "set `trie_cache_capacity: 90001`" is nice and simple. "set `trie_cache_capacities: [[{"verison": 1, "shard_id": 3}, 9000]]`" is way less user friendly.

Comment (matklad) [reply core/store/src/config.rs:100]: Also, did we actually shipped this? Should this be reveted on the next-release branch that? 

If we didn't completely ship the `trie_cache_capacities: [[{"verison": 1, "shard_id": 3}, 9000]]` config, and if it's not mandatory for fixing immediate problems, I would petition to remove the field in this form from the config completely, and spend some time shipping friendly and usable config. 

The specific thing I don't like is the config format -- it's just `derive(Serialize)`, which leads to quite user-unfriendly JSON.

Comment (mina86) [reply core/store/src/config.rs:100]: > Also, did we actually shipped this? Should this be reveted on the next-release branch that?

I’m going to cherry-pick this for 1.28 once this merges.

**Comment by matklad** [comment]
>I don't think we have time right now to refactor trie_cache_capacities, and it seems valuable to me to keep at least some logic in the code instead of understanding later how to revive it in the updated code.

The tradeoff here is that, if we keep this "some logic" and it gets into the release, we'd have to maintain backwards compatability with config format. That's why I personally have a strong preference the "some logic" to be: 

> There's a scalar numeric field in the config which defines default cache size

This seems like a much easier thing to support in a config than the current format with map-with-struct-keys-encoded-as-an-array.

**Review by matklad — APPROVED** [review]

**Pull request merged**